packages/opentelemetry-api/src/api/context.ts

```diff
@@ -0,0 +1,79 @@
+/*!
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  ScopeManager,
+  NoopScopeManager,
+  Context,
+} from '@opentelemetry/scope-base';
+
+/**
+ * Singleton object which represents the entry point to the OpenTelemetry Context API
+ */
+export class ContextAPI {
+  private static _instance?: ContextAPI;
+  private _scopeManager: ScopeManager = new NoopScopeManager();
+
+  /** Empty private constructor prevents end users from constructing a new instance of the API */
+  private constructor() {}
+
+  /** Get the singleton instance of the Scope API */
+  public static getInstance(): ContextAPI {
+    if (!this._instance) {
+      this._instance = new ContextAPI();
+    }
+
+    return this._instance;
+  }
+
+  /**
+   * Set the current context manager. Returns the initialized context manager
+   */
+  public initGlobalContextManager(scopeManager: ScopeManager): ScopeManager {
```

Comment:
should this be named as `setGlobalContextManager` instead of `init...` as it doesn't init anything ? and similar to description `Returns the initialized context manager` 
maybe just `Set the current context manager and then return it.` ?

Reply:
All the API's currently use `init`, and none of them actually init anything. For now I want to keep them consistent. Maybe we should open an issue to rename `init` to `set` and get people's opinions https://github.com/open-telemetry/opentelemetry-js/issues/803